components/merkledb/src/backends/temporarydb.rs

```diff
@@ -14,74 +14,67 @@
 
 //! An implementation of `TemporaryDB` database.
 
-use rocksdb::{WriteBatch, WriteOptions};
-use std::sync::Arc;
-use tempfile::TempDir;
+use crossbeam::sync::ShardedLock;
+use std::{
+    collections::{btree_map::Range, BTreeMap, HashMap},
+    iter::{Iterator, Peekable},
+    ptr,
+    sync::Arc,
+};
 
 use crate::{
-    backends::rocksdb::RocksDBSnapshot, db::DB_METADATA, Database, DbOptions, Iter, Patch,
-    ResolvedAddress, Result, RocksDB, Snapshot,
+    backends::rocksdb::{next_id_bytes, ID_SIZE},
+    db::{check_database, Change, Iterator as DbIterator},
+    Database, Iter, Patch, ResolvedAddress, Result, Snapshot,
 };
 
-/// A wrapper over the `RocksDB` backend which stores data in the temporary directory
-/// using the `tempfile` crate.
-///
-/// This database is only used for testing and experimenting; is not designed to
+type MemoryDB = HashMap<String, BTreeMap<Vec<u8>, Vec<u8>>>;
```

Comment:
Did you try this variant?

```rust
type MemoryDB = HashMap<ResolvedAddress, BTreeMap<Vec<u8>, Vec<u8>>>;
```

Intuitively, it could be more performant since we don't need iteration over multiple indexes in the same family.

Reply:
Yeah. It works good. Thanks for advice 👍🏻